### PR TITLE
Fix Broker Controller discovery w/ LocalLogManager

### PR DESCRIPTION
**Pull request by aloknnikhil**
Verified by running

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/Kip500Broker.scala [hunk 1]
```diff
@@ -111,7 +111,10 @@ class Kip500Broker(val config: KafkaConfig,
   var quotaManagers: QuotaFactory.QuotaManagers = null
 
   val brokerMetaPropsFile = "meta.properties"
-  val brokerMetadataCheckpoints = config.logDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
+
+  // Look for metadata checkpoints (meta.properties) in all log.dirs and metadata.log.dir
+  val brokerMetadataLogDirs = config.logDirs:+(config.metadataLogDir)
```
Comment (cmccabe) [inline core/src/main/scala/kafka/server/Kip500Broker.scala:116]: I'd really prefer just to use an "if" statement here.  The :+ operator is pretty obscure and hard to call up documentation for

File: core/src/main/scala/kafka/server/Kip500Broker.scala [hunk 2]
```diff
@@ -111,7 +111,10 @@ class Kip500Broker(val config: KafkaConfig,
...
+  val brokerMetadataCheckpoints = brokerMetadataLogDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
```
Comment (cmccabe) [inline core/src/main/scala/kafka/server/Kip500Broker.scala:117]: metadata checkpoints appear in all directories, not just log directories

File: core/src/main/scala/kafka/server/Kip500Broker.scala [hunk 3]
```diff
@@ -540,7 +543,7 @@ class Kip500Broker(val config: KafkaConfig,
     val brokerMetadataSet = mutable.HashSet[MetaProperties]()
     val offlineDirs = mutable.ArrayBuffer.empty[String]
 
-    for (logDir <- config.logDirs) {
+    for (logDir <- brokerMetadataLogDirs) {
```
Comment (cmccabe) [inline core/src/main/scala/kafka/server/Kip500Broker.scala:546]: we need to use all log directories, not just the metadata ones

File: core/src/main/scala/kafka/server/Kip500Broker.scala (hunk 2 above)
Comment (aloknnikhil) [reply core/src/main/scala/kafka/server/Kip500Broker.scala:117]: Hmm. I just appended `config.metadataLogDir` to the existing list of directories (which are just `config.logDirs`). Are you saying there's more to add here? I'm not sure what 'all directories' here means.

File: core/src/main/scala/kafka/server/Kip500Broker.scala (hunk 1 above)
Comment (aloknnikhil) [reply core/src/main/scala/kafka/server/Kip500Broker.scala:116]: Yea, fair enough.

File: core/src/main/scala/kafka/server/Kip500Broker.scala (hunk 2 above)
Comment (aloknnikhil) [reply core/src/main/scala/kafka/server/Kip500Broker.scala:117]: Ah. It's the name of the variable I see. Right, I'll fix that.

File: core/src/main/scala/kafka/server/Kip500Broker.scala [hunk 4]
```diff
@@ -111,7 +111,17 @@ class Kip500Broker(val config: KafkaConfig,
   var quotaManagers: QuotaFactory.QuotaManagers = null
 
   val brokerMetaPropsFile = "meta.properties"
-  val brokerMetadataCheckpoints = config.logDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
+
+  // Look for metadata checkpoints (meta.properties) in all log.dirs and metadata.log.dir
+  val brokerLogDirs = config.metadataLogDir match {
```
Comment (cmccabe) [inline core/src/main/scala/kafka/server/Kip500Broker.scala:116]: Hmm, I still don't think this is quite right.  How about a HashSet or something to make sure there are no duplicates?

Comment (aloknnikhil) [reply core/src/main/scala/kafka/server/Kip500Broker.scala:116]: Yea, I agree. That's cleaner. Pushing a change

**Comment by cmccabe** [comment]
LGTM

**Pull request merged**